kb/cryptography/asn1-key-structures-in-der-and-pem.md

```diff
@@ -151,3 +156,33 @@ Within the base64 encoded data, the following DER structure is present:
 ```
 
 The `EncryptedData OCTET STRING` is a PKCS#8 `PrivateKeyInfo`, as described earlier.
+
+## Useful tools
+
+### OpenSSL
+
+```
+openssl asn1parse -inform DER
+```
+
+Limitations:
+
+* Gives up on invalid ASN.1.
+* Gives up on unrecognized OIDs (you can use an [OID file](https://manned.org/asn1parse.1ssl#head5)).
```

Comment:
I don't understand "you can use an OID file" with the link, which doesn't include "OID" anywhere on the page

Reply:
In the Old Days™, when small furry creatures were Real small furry creatures yadda yadda, the `openssl-asn1parse` man page was called `asn1parse`.